packages/extension/src/newtab/KeepItOverlay.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { ReactElement, useCallback, useEffect, useRef } from 'react';
+import {
+  DailyIcon,
+  StraightArrowIcon,
+} from '@dailydotdev/shared/src/components/icons';
+import { IconSize } from '@dailydotdev/shared/src/components/Icon';
+import {
+  Button,
+  ButtonSize,
+  ButtonVariant,
+} from '@dailydotdev/shared/src/components/buttons/Button';
+
+import {
+  Typography,
+  TypographyType,
+} from '@dailydotdev/shared/src/components/typography/Typography';
+import { del } from 'idb-keyval';
+import { useEventListener } from '@dailydotdev/shared/src/hooks';
+import { useLogContext } from '@dailydotdev/shared/src/contexts/LogContext';
+import { LogEvent } from '@dailydotdev/shared/src/lib/log';
+import { INSTALLATION_STORAGE_KEY } from '../lib/common';
+
+interface KeepItOverlayProps {
+  onClose: () => void;
+}
+
+export function KeepItOverlay({ onClose }: KeepItOverlayProps): ReactElement {
+  const { logEvent } = useLogContext();
+  const timeoutRef = useRef<ReturnType<typeof window.setTimeout>>();
+  const onDelete = useCallback(() => del(INSTALLATION_STORAGE_KEY), []);
+
+  const onClick = useCallback(async () => {
+    if (timeoutRef.current) {
+      clearTimeout(timeoutRef.current);
+    }
+
+    logEvent({ event_name: LogEvent.DismissNewTabPermission });
+    await onDelete();
+    onClose();
+  }, [onDelete, onClose, logEvent]);
+
+  // when this overlay is opened, there is nothing the user can do, clicking anywhere should close it.
+  useEventListener(globalThis?.document, 'click', onClick);
+
+  useEffect(() => {
+    onDelete();
+  }, [onDelete]);
```

Comment:
What's this?

Reply:
This is to delete the cache once this opens. On the second reload, they have kept the extension already, without this, there could be a possibility of showing the overlay again.